enterprise/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_3/runtime/interpreted/expressions/NodeProperty.scala

```diff
@@ -34,3 +34,20 @@ case class NodeProperty(offset: Int, token: Int) extends Expression {
 
   override def symbolTableDependencies: Set[String] = Set.empty
 }
+
+case class NodePropertyLate(offset: Int, propKey: String) extends Expression {
+
+  override def apply(ctx: ExecutionContext)(implicit state: QueryState): Any = {
+    val maybeToken = state.query.getOptPropertyKeyId(propKey)
+    if (maybeToken.isEmpty)
```

Comment:
We should look at the produced byte code. 
In these tight loops, I am not after simple code. I'm after fast code. My concern with the `map()` alternative is that we would be creating another `Some` object.